### PR TITLE
normalize master fp to lowercase

### PR DESCRIPTION
**Pull request by scgbckbone**

### CODE REVIEW

File: hwilib/commands.py [hunk 1]
```diff
@@ -114,7 +114,13 @@ def enumerate(password: str = "") -> List[Dict[str, Any]]:
     for module in all_devs:
         try:
             imported_dev = importlib.import_module('.devices.' + module, __package__)
-            result.extend(imported_dev.enumerate(password)) # type: ignore
+            partial_res = imported_dev.enumerate(password)  # type: ignore
+            # normalize master fingerprint to lowercase
+            for device_dict in partial_res:
+                # if None or other type than string - skip
+                if device_dict['fingerprint'] and isinstance(device_dict['fingerprint'], str):
```
Comment (scgbckbone) [inline hwilib/commands.py:121]: not sure if it should raise `ValueError` if master fp is not string

Comment (scgbckbone) [reply hwilib/commands.py:121]: or even attempt some further normalization? , i.e. `if isinstance(device_dict['fingerprint'], bytes):  device_dict['fingerprint'] = device_dict['fingerprint'].hex()`

**Comment by scgbckbone** [comment]
HWI seems to lack some kind of validation layer - probably on purpose left to device developers

**Comment by achow101** [comment]
IMO `commands.py` is the wrong place to put this. As each device implementation is ostensibly part of the public API, each device's `enumerate` should do the normalization.

However, I don't see which device would return an uppercase fingerprint. They all use `get_master_fingerprint` which always returns bytes, and that is always converted to lowercase hex with the python `bytes.hex()` builtin.

**Comment by achow101** [comment]
Concept NACK. Fundamentally, we shouldn't care about the case-ness and everything should be a comparison of the bytes that are represented by the string.

**Comment by scgbckbone** [comment]
I updated [PR](https://github.com/bitcoin/bitcoin/pull/25019) in core

**Comment by scgbckbone** [comment]
closing this as it is meh, does not seem as good idea as yesterday - core should be able to handle both upper/lower case

**Pull request closed without merging**